source/MaterialXView/Viewer.cpp

```diff
@@ -2124,15 +2147,15 @@ void Viewer::bakeTextures()
         // Compute baking resolution.
         mx::ImageVec imageVec = _imageHandler->getReferencedImages(doc);
         auto maxImageSize = mx::getMaxDimensions(imageVec);
-        unsigned int bakeWidth = std::max(maxImageSize.first, (unsigned int) 4);
-        unsigned int bakeHeight = std::max(maxImageSize.second, (unsigned int) 4);
+        unsigned int bakeWidth = std::max(maxImageSize.first, _minimumBakeDimension);
+        unsigned int bakeHeight = std::max(maxImageSize.second, _minimumBakeDimension);
         if (_bakeWidth)
         {
-            bakeWidth = std::max(_bakeWidth, (unsigned int) 4);
+            bakeWidth = std::max(_bakeWidth, _minimumBakeDimension);
```

Comment:
What is the additional value provided by this `minimumBakeDimension` input?

As things currently stand, we have `bakeWidth` and `bakeHeight` values that the user can set, and I believe we would want to use these literal values when they have been provided (e.g. 256x256), rather than clipping them to another minimum (e.g. 1024).

In the alternative case where the user _doesn't_ set `bakeWidth` or `bakeHeight`, then we currently use the maximum size of all input images, falling back to rendering 4x4 images if no input images are present.  In that alternative case, we similarly don't want to clip to an additional minimum size (e.g. 1024), as the user would have provided an explicit `width` and `height` if one was intended.

Reply:
It's been a long while but I think what I encountered is that the `bakeWidth` and `bakeHeight` are not setting from the UI so can end up with very small textures if there are no images mapped as this is the overriding heuristic to "guess" the output image size. 

For example, if you want to bake the marble example you end up with something that is pretty well "useless". 

Perhaps a resolution for this is if `bakeWidth` or `bakeHeight` are explicitly specified, don't use this minimum value and don't use the heuristics value as that is also "deceiving" in that if there are textures mapped then they unknowingly supersede the explicit width and height passed in ?

The alternative is to expose `bakeWidth` and `bakeHeight` which would allow editing of `_bakeWidth` and `_bakeHeight` ? This seems the easist way and adds no new logic.

Reply:
We could indeed add `bakeWidth` and `bakeHeight` as options in the MaterialX viewer, though for production texture baking we usually recommend the use of Python scripts, which already have access to these features.

It's a fine line for the MaterialX viewer, where we want to enable easy testing of features such as texture baking and shader translation, but we don't necessarily want to expose the entire production-level feature set in the viewer interface.

Reply:
I'll leave it out. It's not directly related to the main change anways.